### PR TITLE
MGMT-2417 waiting for controller pod to start running

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: src/common/common.go [hunk 1]
```diff
@@ -42,3 +44,20 @@ func SetConfiguringStatusForHosts(client inventory_client.InventoryClient, inven
...
 	}
 }
+
+func IsPodInStatus(k8Client k8s_client.K8SClient, podNamePrefix string, namespace string, labelMatch map[string]string, status v1.PodPhase, log logrus.FieldLogger) bool {
+	pods, err := k8Client.GetPods(namespace, labelMatch)
+	if err != nil {
+		log.WithError(err).Warnf("Failed to get pod with prefix %s in namespace %s", podNamePrefix, namespace)
+		return false
+	}
+	for _, pod := range pods {
```
Comment (eranco74) [inline src/common/common.go:54]: You expect more than 1?

File: src/installer/installer.go [hunk 2]
```diff
@@ -276,6 +284,30 @@ func (i *installer) waitForBootkube(ctx context.Context) {
...
+		return err
+	}
+	kc, err := i.kcBuilder(KubeconfigPath, i.log)
+	if err != nil {
+		i.log.WithError(err).Errorf("Failed to create kc client from %s", KubeconfigPath)
+		return err
+	}
+	controllerTimeout := time.Duration(i.Config.InstallationTimeout) * time.Minute
+	predicate := func() bool {
+		return common.IsPodInStatus(kc, assistedControllerPrefix, "assisted-installer",
```
Comment (eranco74) [inline src/installer/installer.go:301]: Move assisted-installer to const

File: src/installer/installer.go [hunk 3]
```diff
@@ -276,6 +284,30 @@ func (i *installer) waitForBootkube(ctx context.Context) {
...
+	if err != nil {
+		i.log.WithError(err).Error("Failed to reload resolv.conf")
+		return err
+	}
+	kc, err := i.kcBuilder(KubeconfigPath, i.log)
+	if err != nil {
+		i.log.WithError(err).Errorf("Failed to create kc client from %s", KubeconfigPath)
+		return err
+	}
+	controllerTimeout := time.Duration(i.Config.InstallationTimeout) * time.Minute
```
Comment (eranco74) [inline src/installer/installer.go:299]: timeout seems too long,
How long do you think we should wait?

File: src/ops/ops.go [hunk 4]
```diff
@@ -385,3 +386,26 @@ func (o *ops) UploadInstallationLogs(isBootstrap bool) (string, error) {
 	}
 	return o.ExecPrivilegeCommand(o.logWriter, command, args...)
 }
+
+func (o *ops) ReloadHostFile(filepath string) error {
```
Comment (eranco74) [inline src/ops/ops.go:390]: Add a comment about why this is required

File: src/common/common.go [hunk 5]
```diff
@@ -42,3 +44,20 @@ func SetConfiguringStatusForHosts(client inventory_client.InventoryClient, inven
 		}
 	}
 }
+
+func IsPodInStatus(k8Client k8s_client.K8SClient, podNamePrefix string, namespace string, labelMatch map[string]string, status v1.PodPhase, log logrus.FieldLogger) bool {
+	pods, err := k8Client.GetPods(namespace, labelMatch)
+	if err != nil {
+		log.WithError(err).Warnf("Failed to get pod with prefix %s in namespace %s", podNamePrefix, namespace)
```
Comment (eranco74) [inline src/common/common.go:51]: This will spam the log with errors no?

File: src/common/common.go (hunk 1 above)
Comment (tsorya) [reply src/common/common.go:54]: it is common , yes it is possible for sure

File: src/installer/installer.go (hunk 3 above)
Comment (tsorya) [reply src/installer/installer.go:299]: i think we will fail on timeout in assisted-service earlier. Didn't thought it is really big issue. If everything is ok we will have it running before bootkube finishes

File: src/common/common.go (hunk 5 above)
Comment (tsorya) [reply src/common/common.go:51]: possible but u don't want to see that u don't have a pod when u are trying to reach it?

Comment (tsorya) [reply src/common/common.go:51]: and for sure it must not happened

File: src/installer/installer.go (hunk 3 above)
Comment (tsorya) [reply src/installer/installer.go:300]: but if u think we need a smaller one, lets change just through the number

File: src/common/common.go (hunk 1 above)
Comment (eranco74) [reply src/common/common.go:54]: OK, consider logging all the pods you find with this selector just so we will see what is going on.

File: src/installer/installer.go (hunk 3 above)
Comment (eranco74) [reply src/installer/installer.go:300]: If we expect it to run before bootkube finish, (and we call this function once bootkube is done) let's give it 5 minutes.

Comment (tsorya) [reply src/installer/installer.go:300]: sure, will change

File: src/common/common.go (hunk 1 above)
Comment (tsorya) [reply src/common/common.go:54]: didn't wanted to spawn too much logs

**Review by eranco74 — COMMENTED** [review]
/lgtm
Please move timeout and interval to consts next pr

**Pull request merged**